core/trino-main/src/main/java/io/trino/execution/SqlQueryManager.java

```diff
@@ -241,9 +241,11 @@ public void createQuery(QueryExecution queryExecution)
             throw new TrinoException(GENERIC_INTERNAL_ERROR, format("Query %s already registered", queryExecution.getQueryId()));
```

Comment:
> Final query info notification is not guaranteed to be fired for FAILED state

Why?

Reply:
Final query info is Optional.empty then notification is not fired.  I observed query state is FAILED and final query info is empty, but do not know how to reproduce it.